### PR TITLE
Allow passing of connection

### PR DESCRIPTION
**Pull request by JaTochNietDan**
This is a simple change which shouldn't change any existing implementation (backwards compatible) but it should allow you to use a new function to specify your own TCP dialer for the SMTP client. This allows us to start using a SOCKS5 proxy as discussed before for making SMTP connections from Desk so we can use one IP to send all emails and tell our customers to whitelist this IP address via CNAME or otherwise.

### CODE REVIEW

File: email.go [hunk 1]
```diff
@@ -211,31 +221,38 @@ func (e *Email) Send(addr string, a smtp.Auth, TLSMandatory bool) error {
...
+func sendMailUsingConn(
+	conn net.Conn,
+	addr string,
+	from string,
+	to []string,
+	msg []byte,
+	TLSMandatory bool,
+	a smtp.Auth,
+) error {
+	host, _, _ := net.SplitHostPort(addr)
```
Comment (ready4god2513) [inline email.go:250]: Are we guaranteed that the address is a valid format right here?  If not, we should check the error.

Comment (JaTochNietDan) [reply email.go:250]: See line 56 of stdlib, I'm just copying their logic.

https://golang.org/src/net/smtp/smtp.go#56

Comment (ready4god2513) [reply email.go:250]: stdlib is bad.

Comment (ready4god2513) [reply email.go:250]: I guess the next line would fail anyways if this fails.

**Review by ready4god2513 — APPROVED** [review]

**Pull request merged**